internal/cmd/mongodbflex/backup/list/list.go

```diff
@@ -0,0 +1,169 @@
+				return err
+			}
+
+			instanceLabel, err := mongodbflexUtils.GetInstanceName(ctx, apiClient, model.ProjectId, *model.InstanceId)
+			if err != nil {
+				p.Debug(print.ErrorLevel, "get instance name: %v", err)
+				instanceLabel = *model.InstanceId
+			}
+
+			// Call API
+			req := buildRequest(ctx, model, apiClient)
+			resp, err := req.Execute()
+			if err != nil {
+				return fmt.Errorf("get backups for MongoDB Flex instance %q: %w", instanceLabel, err)
+			}
+			if resp.Items == nil || len(*resp.Items) == 0 {
+				cmd.Printf("No backups found for instance %q\n", instanceLabel)
+				return nil
+			}
+			backups := *resp.Items
+
+			// Truncate output
+			if model.Limit != nil && len(backups) > int(*model.Limit) {
+				backups = backups[:*model.Limit]
+			}
+
+			return outputResult(ctx, p, model, apiClient, backups)
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Var(flags.UUIDFlag(), instanceIdFlag, "Instance ID")
+	cmd.Flags().Int64(limitFlag, 0, "Maximum number of entries to list")
+
+	err := flags.MarkFlagsRequired(cmd, instanceIdFlag)
+	cobra.CheckErr(err)
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command) (*inputModel, error) {
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	limit := flags.FlagToInt64Pointer(p, cmd, limitFlag)
+	if limit != nil && *limit < 1 {
+		return nil, &errors.FlagValidationError{
+			Flag:    limitFlag,
+			Details: "must be greater than 0",
+		}
+	}
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		InstanceId:      flags.FlagToStringPointer(p, cmd, instanceIdFlag),
+		Limit:           flags.FlagToInt64Pointer(p, cmd, limitFlag),
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *mongodbflex.APIClient) mongodbflex.ApiListBackupsRequest {
+	req := apiClient.ListBackups(ctx, model.ProjectId, *model.InstanceId)
+	return req
+}
+
+func outputResult(ctx context.Context, p *print.Printer, model *inputModel, apiClient *mongodbflex.APIClient, backups []mongodbflex.Backup) error {
+	switch model.OutputFormat {
+	case print.JSONOutputFormat:
+		details, err := json.MarshalIndent(backups, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal MongoDB Flex backups list: %w", err)
+		}
+		p.Outputln(string(details))
+
+		return nil
+	default:
+		table := tables.NewTable()
+		table.SetHeader("ID", "CREATED AT", "EXPIRES AT", "BACKUP SIZE", "RESTORE STATUS")
+		for i := range backups {
+			backup := backups[i]
+			restoreStatus, err := mongodbflexUtils.GetRestoreStatus(ctx, apiClient, model.ProjectId, *model.InstanceId, *backup.Id)
```

Comment:
It could lead to high response times on the command if there is a high volume of backups and restores, but if we want to show the restore status for each backup it seems like the only way.

I tried another alternative by making `GetRestoreStatus` receive a list of backups instead of a single one and returning a map, but it turned out much more complex and with no significant improve on the performance (we still had to iterate over two arrays.

The current implementation also forces changing the signature of the usual `outputResult`, unfortunately

Reply:
Wouldn't it be possible to get all the restore jobs for the instance once and then have the utils function receive the list of jobs?
Because the largest overhead in our commands is usually the API call

Reply:
Thanks for the suggestion, I simplified the code a lot and saved quite some requests